resteasy-client/src/main/java/org/jboss/resteasy/plugins/providers/sse/client/SseEventSourceImpl.java

```diff
@@ -255,10 +258,9 @@ public boolean close(final long timeout, final TimeUnit unit)
       internalClose();
       try
       {
-         if (!executor.awaitTermination(timeout, unit))
-         {
-            return false;
-         }
+         // mimic executor.awaitTermination giving time for tasks to complete
+         long duration = TimeUnit.MILLISECONDS.convert(timeout, unit);
+         Thread.sleep(duration);
```

Comment:
Is it possible to run internalClose() with schduleExecutor.scheduleWithFixedDelay() without thread.sleep?   Is it better ?